### ISSUE TITLE
Detect if page is already saved and highlight icon

### ISSUE THREAD

**Issue by alexkrolick** [Pocket/extension-save-to-pocket#209]
## Expected Behavior

Feature request: if a page is already saved in Pocket, use the Saved Article state for the address bar icon (ie red instead of gray); or, at least detect the duplicate at save time and warn about it

## Actual Behavior

When coming back to a page that I already saved, the icon is the standard gray color, so I sometimes save the page again and create duplicates
## Specifications

* Version: All
* Browser: All
* Operating System: All

**Comment by collectedmind** [comment]
@alexkrolick The main challenge here is using bandwidth behind the scenes without intent.  With mild usage this would have less of an impact.  Over time we would be making a request on every tab opened and it would start to add up for the user in terms of bandwidth impact.  The other side of the coin are extremely large lists and downloading that data to the users computer would start to eat up memory without intent.

I do agree this would be an excellent addition to the extension, so perhaps we can find some middle ground here.

**Comment by alexkrolick** [comment]
The mobile appears to be able to detect this. It says "This page is already saved. Moving it to the top of your queue." What's the difference with the extension?

**Comment by collectedmind** [comment]
@alexkrolick Yep! It is the case that the mobile apps use an offline store.  We have spoken about ways to implement this on web/extension and I think there's some options to explore.  I have played around this idea with web-workers and IndexedDB (which presented some additional challenges due to the nature of extensions and IndexedDB support).  

All in all this is something that I think would make a good addition and as we start to revisit and prioritize I will keep this in the hopper.
